### DISCUSSION TITLE
[2024-09-02] Library & Engine Team Update

### DISCUSSION THREAD

**Discussion by GregoryTravis** [enso-org/enso#10943] [Weekly Demo Day]
- James: Tuning S3_File doc comments. #10832
- James: Wip/jd/fast excel read #10877
- James: Update what needs to be re-signed. #10922
- James: Small tweaks from QA #10941
- Radosław: Fix `Count_Distinct` in Snowflake #10818
- Radosław: Experiment: Reporting test failures as annotations #10821
- Radosław: `First`/`Last` support in Snowflake, and partially in SQLite #10841
- Radosław: Move Connection classes out of Internal #10886
- Radosław: Include `projectName` in audit logs #10892
- Radosław: Pending Audit Logs are sent in batches #10918
- Radosław: Override CWD in `runEngineDistribution` to the parent of the project being run to avoid warning #10928
- Greg: Column-level lexically-scoped CTE expressions #10826
- Greg: Add Text.to_decimal #10874
- Greg: Implement .floor, .ceil, .trunc for the in-memory Decimal column #10887
- Greg: Upgrade SQLite to 3.46 #10911
- Greg: Add vectorized rounding operation to Decimal columns #10912
- Ad: Wip/adr/more sql server tests #10810
- Martha: Add drilldown to XML #10824
- Martha: Sort toolbar rendering for narrow nodes #10831
- Martha: Initial enso changes to ScatterPlot #10871
- Martha: text format selection bug #10940
- Hubert: Dockerize NodeJS-based Ydoc #10811
- Hubert: Avoid copying IR nodes when no change is performed in passes #10839
- Hubert: Ensure clients get write capability #10872
- Hubert: Enable IR caches in Language Server #10880
- Hubert: Temporary workaround for #10921 #10931 (disables caching)
- Hubert: Ensure execution is scheduled after compilation #10883
- Hubert: Add a test demonstrating compilation/execution sequence #10898
- Hubert: Akka-related startup improvements #10905
- Hubert: Update reflect-config.json to accomodate for akka changes #10930 (follow up on #10905)
- Pavel: engine-runner and language-server are separate JPMS modules #10823
- Pavel: Do not run enterprise engine and stdlib benchmarks #10829
- Pavel: Simplify declaration of module path in sbt build #10836
- Pavel: Fix std benchmarks run #10902
- Dmitry: Lazy Diagnostic storage allocation #10852
- Dmitry: Provide missing arguments when stepping into a function #10879
- Dmitry: Add CI checks for aarch64 #10881
- Dmitry: Fix S3 spec #10893
- Dmitry: Execute program when required after updating IdMap #10939

**Comment by JaroslavTulach** [comment]
![My Oracle Support request](https://github.com/user-attachments/assets/a336a2db-1129-407b-817d-0f8a9f40c0f1)

- Requested [backport of GraalVM fix](https://github.com/oracle/graal/pull/9516#issuecomment-2311796280) into GraalVM for JDK21
- Print out warnings associated with local variables #10842
- `IO.print` without new line #10858
- 
![folds](https://github.com/user-attachments/assets/05b15747-8594-494a-9409-d7eeaaff68c2)

- Fold output of each benchmark #10914
- Method return type checks avoid conversion when the type fits #10882
- When there are no `pendingItems` for a `KEEP_ALIVE` period, stop RequestQueue observer thread #10890
- Speeding up RuntimeVisualizationsTest by one minute #10891


![2s](https://github.com/user-attachments/assets/04b95886-7f16-429c-a0f7-418fb01fe764)
![1s](https://github.com/user-attachments/assets/63627ad7-f065-43e8-9ed4-a733819e41d9)

- Benchmark and speed processing of polyglot java imports up #10899

![still 10%](https://github.com/user-attachments/assets/d62eb95f-a1a8-42b2-be31-482f7cdd0607)

- Delay loading of `AliasAnalysis` data #10837
- Provide names of local variables via `FramePointerAnalysis` #10906
